### PR TITLE
one valuefetcher refactoring

### PR DESCRIPTION
**Pull request by andimarek**

### CODE REVIEW

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 1]
```diff
@@ -208,6 +206,62 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
...
+            }
+
+            @Override
+            protected void handleResult(Object value, Throwable throwable) {
+                fetchCtx.onCompleted(value, throwable);
+            }
+        };
+        ExecutionStepInfoFactory executionStepInfoFactory = new ExecutionStepInfoFactory(executionContext);
+        ExecutionStepInfo subStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(parameters.getField(), parameters.getExecutionStepInfo());
+        return valueFetcher
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java]: I still think `ExecutionStepInfoFactory` should be a singleton that takes the per request things (execitionCOntext et al) as parameters

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 2]
```diff
@@ -222,102 +276,82 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
      *
      * @throws NonNullableFieldWasNullException in the future if a non null field resolves to a null value
      */
-    protected CompletableFuture<Object> fetchField(ExecutionContext executionContext, ExecutionStrategyParameters parameters) {
-        Field field = parameters.getField().get(0);
-        GraphQLObjectType parentType = (GraphQLObjectType) parameters.getExecutionStepInfo().getUnwrappedNonNullType();
-        GraphQLFieldDefinition fieldDef = getFieldDef(executionContext.getGraphQLSchema(), parentType, field);
+//    protected CompletableFuture<Object> fetchField(ExecutionContext executionContext, ExecutionStrategyParameters parameters) {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java]: do you want this commented out code??

File: src/main/java/graphql/execution/ValueFetcher.java [hunk 3]
```diff
@@ -32,13 +27,17 @@
 import java.util.concurrent.CompletionStage;
 import java.util.stream.Collectors;
 
+import static graphql.Assert.assertNotNull;
 import static graphql.schema.DataFetchingEnvironmentBuilder.newDataFetchingEnvironment;
+import static java.util.Collections.singletonList;
 
 @Internal
 public class ValueFetcher {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ValueFetcher.java]: JavaDoc on its reason for being

**Comment by andimarek** [comment]
I an not really happy with that so far. Will not merge it for now

**Pull request closed without merging**